server/db/postgres/adapter.go

```diff
@@ -0,0 +1,2695 @@
+		return err
+	}
+
+	fmt.Println("creating dellog")
+	// Deletion log
+	if _, err = tx.Exec(
+		`CREATE TABLE dellog(
+			id         SERIAL,
+			topic      VARCHAR(25) NOT NULL,
+			deletedfor BIGINT NOT NULL DEFAULT 0,
+			delid      INT NOT NULL,
+			low        INT NOT NULL,
+			hi         INT NOT NULL,
+			PRIMARY KEY(id),
+			FOREIGN KEY(topic) REFERENCES topics(name)
+		);`); err != nil {
+		return err
+	}
+	fmt.Println("creating dellog index 1")
+	if _, err = tx.Exec(
+		`CREATE INDEX dellog_topic_delid_deletedfor on dellog(
+			topic, delid, deletedfor
+		)`); err != nil {
+		return err
+	}
+
+	fmt.Println("creating dellog index 2")
+	if _, err = tx.Exec(
+		`CREATE INDEX dellog_topic_deletedfor_low_hi on dellog(
+			topic, deletedfor, low, hi
+		)`); err != nil {
+		return err
+	}
+	fmt.Println("creating dellog index 3")
+
+	if _, err = tx.Exec(
+		`CREATE INDEX dellog_deletedfor on dellog(
+			deletedfor
+		)`); err != nil {
+		return err
+	}
+
+	fmt.Println("creating credentials")
+	// User credentials
+	if _, err = tx.Exec(
+		`CREATE TABLE credentials(
+			id        SERIAL,
+			createdat TIMESTAMP NOT NULL,
+			updatedat TIMESTAMP NOT NULL,
+			deletedat TIMESTAMP,
+			method    VARCHAR(16) NOT NULL,
+			value     VARCHAR(128) NOT NULL,
+			synthetic VARCHAR(192) NOT NULL,
+			userid    BIGINT NOT NULL,
+			resp      VARCHAR(255),
+			done      smallint NOT NULL DEFAULT 0,
+			retries   INT NOT NULL DEFAULT 0,
+			PRIMARY KEY(id),
+			CONSTRAINT credentials_uniqueness UNIQUE(synthetic),
+			FOREIGN KEY(userid) REFERENCES users(id)
+		);`); err != nil {
+		return err
+	}
+
+	// Records of uploaded files.
+	// Don't add FOREIGN KEY on userid. It's not needed and it will break user deletion.
+	if _, err = tx.Exec(
+		`CREATE TABLE fileuploads(
+			id        BIGINT NOT NULL,
+			createdat TIMESTAMP NOT NULL,
+			updatedat TIMESTAMP NOT NULL,
+			userid    BIGINT NOT NULL,
+			status    INT NOT NULL,
+			mimetype  VARCHAR(255) NOT NULL,
+			size      BIGINT NOT NULL,
+			location  VARCHAR(2048) NOT NULL,
+			PRIMARY KEY(id)
+		)`); err != nil {
+		return err
+	}
+
+	// Links between uploaded files and the messages they are attached to.
+	if _, err = tx.Exec(
+		`CREATE TABLE filemsglinks(
+			id			SERIAL,
+			createdat	TIMESTAMP NOT NULL,
+			fileid		BIGINT NOT NULL,
+			msgid 		INT NOT NULL,
+			PRIMARY KEY(id),
+			FOREIGN KEY(fileid) REFERENCES fileuploads(id) ON DELETE CASCADE,
+			FOREIGN KEY(msgid) REFERENCES messages(id) ON DELETE CASCADE
+		)`); err != nil {
+		return err
+	}
+
+	fmt.Println("creating kvmeta")
+	if _, err = tx.Exec(
+		`CREATE TABLE kvmeta(
+			key CHAR(32),
+			value TEXT,
+			PRIMARY KEY(KEY)
+		)`); err != nil {
+		return err
+	}
+	fmt.Println("inserting kvmeta")
+	if _, err = tx.Exec("INSERT INTO kvmeta(key, value) VALUES ('version'," + strconv.Itoa(adpVersion) + ")"); err != nil {
+		return err
+	}
+
+	fmt.Println("creation db complete")
+	return tx.Commit()
+}
+
+func (a *adapter) UpgradeDb() error {
```

Comment:
All you need here is `return nil`. All the code below is unnecessary. Postgress adapter does not exist for older versions and consequently does not need to support the upgrades.

Reply:
Thanks. I will update